tools/main.go

```diff
@@ -110,33 +119,39 @@ func convertConfigImageToRegsyncImages(repo config.Repository, image *config.Ima
 }
 
 func migrateImagesList(_ context.Context, cmd *cli.Command) error {
-	imagesListPath := cmd.Args().Get(0)
-	if imagesListPath == "" {
-		return fmt.Errorf("must pass path to images list file as argument")
+	if cmd.Args().Len() != 2 {
+		return fmt.Errorf("must pass source and target image")
 	}
+	sourceImage := cmd.Args().Get(0)
+	targetImage := cmd.Args().Get(1)
 
 	configYaml, err := config.Parse(configYamlPath)
 	if err != nil {
-		return fmt.Errorf("failed to parse %q: %w", configYamlPath, err)
+		return fmt.Errorf("failed to parse config: %w", err)
 	}
 	accumulator := config.NewImageAccumulator()
 	for _, existingImage := range configYaml.Images {
 		accumulator.AddImage(existingImage)
 	}
 
-	legacyImages, err := legacy.ParseImagesList(imagesListPath)
+	imagesListComment, legacyImages, err := legacy.ParseImagesList(imagesListPath)
 	if err != nil {
-		return fmt.Errorf("failed to parse %q: %w", imagesListPath, err)
+		return fmt.Errorf("failed to parse images list: %w", err)
 	}
 
 	configJson, err := legacy.ParseConfig(configJsonPath)
 	if err != nil {
 		return fmt.Errorf("failed to parse %q: %w", configJsonPath, err)
 	}
 
+	if configJson.Contains(sourceImage) {
+		log.Printf("warning: %s refers to image with source %q", configJsonPath, sourceImage)
+	}
+
+	legacyImagesToKeep := make([]legacy.ImagesListEntry, 0, len(legacyImages))
 	for _, legacyImage := range legacyImages {
-		// if image in config.json, skip
-		if configJson.Contains(legacyImage.Source) {
+		if legacyImage.Source != sourceImage || legacyImage.Target != targetImage {
```

Comment:
I think this would be clearer without the negatives
```
if legacyImage.Source == sourceImage && legacyImage.Target == targetImage {
  newImage, err := convertImageListEntryToImage(legacyImage)
  if err != nil {
    return fmt.Errorf("failed to convert %q: %w", legacyImage, err)
  }
  accumulator.AddImage(newImage)
} else {
  legacyImagesToKeep = append(legacyImagesToKeep, legacyImage)
  continue
}
```